src/compas/files/gltf.py

```diff
@@ -0,0 +1,657 @@
+                accessor_offset,
+                NUM_COMPONENTS_BY_TYPE_ENUM['SCALAR']
+            )
+
+            sparse_values_data = sparse_data['values']
+            sparse_values_buffer_view_index = sparse_values_data['bufferView']
+            sparse_values = self.read_from_buffer_view(
+                sparse_values_buffer_view_index,
+                sparse_count,
+                component_type,
+                accessor_offset,
+                num_components
+            )
+
+            for index, data_index in enumerate(sparse_indices):
+                data[data_index] = sparse_values[index]
+
+            if accessor.get('normalized', False):
+                for index, tuple_ in enumerate(data):
+                    new_tuple = ()
+                    for i in tuple_:
+                        if component_type == COMPONENT_TYPE_BYTE:
+                            new_tuple += (max(float(i / 127.0), -1.0),)
+                        elif component_type == COMPONENT_TYPE_UNSIGNED_BYTE:
+                            new_tuple += (float(i / 255.0),)
+                        elif component_type == COMPONENT_TYPE_SHORT:
+                            new_tuple += (max(float(i / 32767.0), -1.0),)
+                        elif component_type == COMPONENT_TYPE_UNSIGNED_SHORT:
+                            new_tuple += (i / 65535.0,)
+                        else:
+                            new_tuple += (float(i),)
+                    data[index] = new_tuple
+
+        return data
+
+    def get_generic_data(self, num_components, count):
+        return [(0, ) * num_components for _ in range(count)]
+
+    def read_from_buffer_view(self, buffer_view_index, count, component_type, accessor_offset, num_components):
+        buffer_view = self.json['bufferViews'][buffer_view_index]
+
+        buffer_view_offset = buffer_view.get('byteOffset', 0)
+        offset = accessor_offset + buffer_view_offset
+
+        format_char = COMPONENT_TYPE_ENUM[component_type]
+        format_ = '<' + format_char * num_components
+
+        expected_length = struct.calcsize(format_)
+
+        component_size = struct.calcsize('<' + format_char)
+        if component_type == 'MAT2' and component_size == 1:
+            format_ = '<FFxxFF'.replace('F', format_char)
+            expected_length = 8
+        elif component_type == 'MAT3' and component_size == 1:
+            format_ = '<FFFxFFFxFFF'.replace('F', format_char)
+            expected_length = 12
+        elif component_type == 'MAT3' and component_size == 2:
+            format_ = '<FFFxxFFFxxFFF'.replace('F', format_char)
+            expected_length = 24
+
+        byte_stride = buffer_view.get('byteStride', expected_length)
+
+        unpack_from = struct.Struct(format_).unpack_from
+
+        buffer_index = buffer_view['buffer']
+        buffer = self.get_buffer(buffer_index)
+
+        data = [
+            unpack_from(buffer[i: i + byte_stride].tobytes())
```

Comment:
Slicing the memoryview and calling tobytes() is the change I had to make for IronPython.

Reply:
👍